### ISSUE TITLE
UFCS/ struct initializer ambiguity

### ISSUE THREAD

**Issue by DaveDuck321** [DaveDuck321/GrapheneLang#96]
Just got the following error: hadn't even considered it until now. The following is ambiguous:

```javascript
let a = {member : fn()}; // A named struct initializer where the member is set to the result of `fn()`
let a = {member:fn()}; // An unnamed struct initializer where the first member is set to the result of `member:fn()`
```

I think we should just remove the option to have unnamed struct initializers... Thoughs @antroseco.

**Comment by antroseco** [comment]
Guhh, yeah that's annoying. 

> I think we should just remove the option to have unnamed struct initializers

We could, but then it's inconsistent with initializer lists isn't it. And that's a pretty nice feature to have.

What if we change UFCS to `.` instead of `:`?

**Comment by DaveDuck321** [comment]
> We could, but then it's inconsistent with initializer lists isn't it. And that's a pretty nice feature to have.
Yeah true, we'd need to rethink arrays :cry:

> What if we change UFCS to `.` instead of `:`?

That'd work rn but it'll break eventually when we'll want to add callables.... `a.func()` would be ambiguous between UFCS and the callable member `func` of struct `a`... Unless you cant to copy FORTRAN an use `call a.func` for members (which I'm not a fan of tbh).

**Comment by DaveDuck321** [comment]
Oooo, ok... How about we change the named initializer list syntax to match c++?

```
{.a : func()}; // Unambiguously means member a = func()
{a:func()}; // Unambiguosly means first member = a:func()
```

**Comment by antroseco** [comment]
Yeah, that's a good solution

**Comment by DaveDuck321** [comment]
Fixed in 63e3c278f43525ec0b98d862fcb3eb600628fb14
